packages/sit-onyx/src/components/OnyxDataGrid/OnyxDataGrid.stories.ts

```diff
@@ -36,27 +39,47 @@ export const HeaderInteractions = {
         name: Symbol("More actions"),
         watch: [],
         header: {
-          actions: () => [
-            {
-              iconComponent: h(OnyxSystemButton, {
-                label: "Column options",
-                icon: sort,
-                color: "medium",
-              }),
-              listItems: [
-                h(OnyxListItem, () => "Pin column"),
-                h(OnyxListItem, () => "Unpin column"),
-              ],
-            },
-            {
-              iconComponent: h(OnyxSystemButton, {
-                label: "Column options",
-                icon: sort,
-                color: "medium",
-              }),
-              listItems: [h(OnyxListItem, () => "Remove column")],
-            },
-          ],
```

Comment:
Please keep the more generic approach with `iconComponent` and `listItems` render functions. We will need to support more complex use-cases like the search input.